compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -4013,9 +4175,10 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
           } else platform.externalEquals
           // scalastyle:on line.size.limit
         }
-        val moduleClass = equalsMethod.owner
-        val instance = genLoadModule(moduleClass)
-        genApplyMethod(instance, equalsMethod, List(lsrc, rsrc))
+        if (BoxesRunTimeClass.isJavaDefined)
```

Comment:
Hum, when does this happen? Isn't BoxesRunTime in the scalalib?

Reply:
It is in the scalalib, but upstream it is written [in Java](https://github.com/scala/scala/blob/2.13.x/src/library/scala/runtime/BoxesRunTime.java), so under separate compilation it is Java-defined.